controller/work_item_labels_blackbox_test.go

```diff
@@ -0,0 +1,227 @@
+				"version": fixtures.WorkItems[0].Version,
+			},
+			Relationships: &app.WorkItemRelationships{},
+		},
+	}
+	// attach 2 labels
+	apiLabelType := label.APIStringTypeLabels
+	lbl0 := fixtures.Labels[0].ID.String()
+	lbl1 := fixtures.Labels[1].ID.String()
+	u.Data.Relationships.Labels = &app.RelationGenericList{
+		Data: []*app.GenericData{
+			{
+				ID:   &lbl0,
+				Type: &apiLabelType,
+			},
+			{
+				ID:   &lbl1,
+				Type: &apiLabelType,
+			},
+		},
+	}
+	_, updatedWI := test.UpdateWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, &u)
+	assert.NotNil(l.T(), updatedWI)
+	require.NotNil(l.T(), updatedWI.Data.Relationships.Labels.Links)
+	assert.Contains(l.T(), *updatedWI.Data.Relationships.Labels.Links.Related, relatedLink)
+	assert.Len(l.T(), updatedWI.Data.Relationships.Labels.Data, 2)
+	mustHave := map[string]struct{}{
+		lbl0: {},
+		lbl1: {},
+	}
+	for _, lblData := range updatedWI.Data.Relationships.Labels.Data {
+		delete(mustHave, *lblData.ID)
+	}
+	require.Empty(l.T(), mustHave)
+
+	// attach 1 label
+	lbl2 := fixtures.Labels[2].ID.String()
+	u.Data.Attributes["version"] = updatedWI.Data.Attributes["version"]
+	u.Data.Relationships.Labels = &app.RelationGenericList{
+		Data: []*app.GenericData{
+			{
+				ID:   &lbl2,
+				Type: &apiLabelType,
+			},
+		},
+	}
+	_, updatedWI = test.UpdateWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, &u)
+	assert.NotNil(l.T(), updatedWI)
+	require.NotNil(l.T(), updatedWI.Data.Relationships.Labels.Links)
+	assert.Contains(l.T(), *updatedWI.Data.Relationships.Labels.Links.Related, relatedLink)
+	assert.Len(l.T(), updatedWI.Data.Relationships.Labels.Data, 1)
+	mustHave = map[string]struct{}{
+		lbl2: {},
+	}
+	for _, lblData := range updatedWI.Data.Relationships.Labels.Data {
+		delete(mustHave, *lblData.ID)
+	}
+	require.Empty(l.T(), mustHave)
+
+	// detach all labels
+	u.Data.Attributes["version"] = updatedWI.Data.Attributes["version"]
+	u.Data.Relationships.Labels = &app.RelationGenericList{
+		Data: []*app.GenericData{},
+	}
+	_, updatedWI = test.UpdateWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, &u)
+	assert.NotNil(l.T(), updatedWI)
+	assert.Empty(l.T(), updatedWI.Data.Relationships.Labels.Data)
+	require.NotNil(l.T(), updatedWI.Data.Relationships.Labels.Links)
+	assert.Contains(l.T(), *updatedWI.Data.Relationships.Labels.Links.Related, relatedLink)
+
+	// verify distinct labels are attached
+	lbl1 = fixtures.Labels[1].ID.String()
+	lbl2 = fixtures.Labels[2].ID.String()
+	u.Data.Attributes["version"] = updatedWI.Data.Attributes["version"]
+	u.Data.Relationships.Labels = &app.RelationGenericList{
+		Data: []*app.GenericData{
+			{
+				ID:   &lbl1,
+				Type: &apiLabelType,
+			}, {
+				ID:   &lbl2,
+				Type: &apiLabelType,
+			}, {
+				ID:   &lbl2,
+				Type: &apiLabelType,
+			},
+		},
+	}
+	_, updatedWI = test.UpdateWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, &u)
+	assert.NotNil(l.T(), updatedWI)
```

Comment:
`require.NotNil()` here too

Reply:
done in https://github.com/fabric8-services/fabric8-wit/pull/1627/commits/d669b65ccc5f6d78c3bc11494e6065ee6341559c